engine/Shopware/Components/Translation.php

```diff
@@ -391,6 +428,288 @@ public function write($language, $type, $key = 1, $data = null, $merge = false)
+     *          'name' => 'debit',
+     *
+     *          'description' => 'Debit',
+     *          'additionalDescription' => 'Insert text',
+     *
+     *           // Additional values
+     *          ...
+     *      ]
+     * ];
+     *
+     * @param array  $objects               Objects to be translated
+     * @param string $type                  Objecttype of s_core_translations
+     * @param array  $translationMappingArg How the translation should be mapped onto the objects
+     * @param int    $language              Language id
+     * @param int    $fallback              Id of fallback language
+     *
+     * @return array Translated objects
+     */
+    public function translateObjects($objects, $type, $translationMappingArg, $language = null, $fallback = null)
+    {
+        // Check if the languages are specified and query them if not
+        if ($language === null) {
+            $language = $this->language;
+            $fallback = $this->fallback;
+        }
+
+        $fallback = $fallback ?: $this->getFallbackLocaleId($language);
+
+        // Load translations from db
+        $translations = $this->readBatchWithFallback($language, $fallback, $type);
+
+        // Create correct mapping array from parameter
+        $translationMapping = array_map(function ($value) {
+            if (is_array($value)) {
+                return $value;
+            }
+
+            return [
+                'source' => $value,
+                'destination' => $value,
+            ];
+        }, $translationMappingArg);
+
+        // Map translations onto properties
+        foreach ($objects as $objectKey => $object) {
+            $translationObj = $translations[$object['id']];
+
+            $translatedValue = '';
+
+            // loop through all mappings
+            foreach ($translationMapping as $mapping) {
+                // Check if a translation exists
+                if ($translationObj) {
+                    $keyOfTranslationObj = $mapping['source'];
+                    $translatedValue = $translationObj[$keyOfTranslationObj];
+                } elseif ($mapping['default']) {
+                    $translatedValue = $object[$mapping['default']];
+                }
+
+                // Check if a value to be mapped exists and map it only, if it does exist
+                if ($translatedValue) {
+                    $object[$mapping['destination']] = $translatedValue;
+                }
+            }
+
+            // Save changes
+            $objects[$objectKey] = $object;
+        }
+
+        return $objects;
+    }
+
+    /**
+     * Loads the id of the fallback language
+     *
+     * @param int $currentLocaleId
+     *
+     * @return int
+     */
+    public function getFallbackLocaleId($currentLocaleId)
+    {
+        if ($currentLocaleId === 1) {
+            return 1;
+        }
+
+        $fallback = $this->connection->fetchColumn(
+            "SELECT id FROM s_core_locales WHERE locale = 'en_GB'"
```

Comment:
What happens, if the `en_GB` locale does not exist?

Reply:
Added a fallback for the fallback to German 😉